dbms/src/DataStreams/PushingToViewsBlockOutputStream.cpp

```diff
@@ -73,35 +76,30 @@ void PushingToViewsBlockOutputStream::write(const Block & block)
     if (replicated_output && replicated_output->lastBlockIsDuplicate())
         return;
 
-    /// Insert data into materialized views only after successful insert into main table
-    for (auto & view : views)
+    // Insert data into materialized views only after successful insert into main table
+    bool allow_concurrent_view_processing = context.getSettingsRef().allow_concurrent_view_processing;
+    if (allow_concurrent_view_processing && views.size() > 1)
     {
-        try
+        // Push to views concurrently if enabled, and more than one view is attached
+        ThreadPool pool(std::min(getNumberOfPhysicalCPUCores(), views.size()));
```

Comment:
Why don't use `settings.max_threads`?

Reply:
I can change it if you'd prefer that

Reply:
I've already done it but forgot to commit.